database/dialects.go

```diff
@@ -32,7 +34,46 @@ const (
 	DialectVertica Dialect = "vertica"
 )
 
-// NewStore returns a new [Store] implementation for the given dialect.
+// ParseDialect returns the corresponding [Dialect], if any, for a given string with external
+// origin. A supported [Dialect] value, like [DialectPostgres], may have multiple supported
+// aliases, e.g. "postgres" or "pgx".
+func ParseDialect(s string) (d Dialect, err error) {
+	// Every non-error return value from this function must have an entry in the [NewStore] lookup map.
+	switch s {
+	case "postgres", "pgx":
+		d = DialectPostgres
+	case "mysql":
+		d = DialectMySQL
+	case "sqlite3", "sqlite":
+		d = DialectSQLite3
+	case "spanner":
+		d = DialectSpanner
+	case "mssql", "azuresql", "sqlserver":
+		d = DialectMSSQL
+	case "redshift":
+		d = DialectRedshift
+	case "tidb":
+		d = DialectTiDB
+	case "clickhouse":
+		d = DialectClickHouse
+	case "vertica":
+		d = DialectVertica
+	case "ydb":
+		d = DialectYdB
+	case "turso":
+		d = DialectTurso
+	case "starrocks":
+		d = DialectStarrocks
+	case "dsql":
```

Comment:
This case is new. The lookup map in `NewStore` has an entry for it, so we should support it here as well.